### PR TITLE
Establish compatibility with old module

### PR DESCRIPTION
**Pull request by leonampd**
### Description

The new module wasn't compatible with old one. This change the name of the module itself as well its configuration's names to avoid conflicts. Now, our customers can use both versions at the same time without problems. They can migrate gradually to the new version too.

### Issue

There's no issue

### Tests

We have automated tests that validate all features implemented on new version. For the old version we tested, manually, all transaction workflow with both versions installed and enabled. No errors was found.

### CODE REVIEW

File: app/code/community/PagarMe/Core/etc/config.xml [hunk 1]
```diff
@@ -106,23 +106,22 @@
...
+            <pagarme_configurations>
+                <modal_title>Pagarme Checkout</modal_title>
+                <modal_button_text>Realizar pagamento</modal_button_text>
                 <creditcard_interest_rate>0</creditcard_interest_rate>
                 <creditcard_free_installments>1</creditcard_free_installments>
                 <creditcard_max_installments>1</creditcard_max_installments>
                 <creditcard_allowed_creditcard_brands>visa,mastercard,amex,hipercard,aura,jcb,diners,elo,discover</creditcard_allowed_creditcard_brands>
                 <creditcard_title>Pagamento com cartão de crédito</creditcard_title>
-                <checkout_capture_customer_data>true</checkout_capture_customer_data>
-                <checkout_button_text>Realizar pagamento</checkout_button_text>
```
Comment (devdrops) [inline app/code/community/PagarMe/Core/etc/config.xml]: This option is not available anymore?

File: app/design/frontend/base/default/template/pagarme/form/pagarme_modal.phtml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    <input type="hidden" value="" name="payment[pagarme_modal_interest_rate]" id="pagarme_modal_interest_rate" />
+
+    <div style="display:none;" id="advice-required-entry-pagarme-modal-hash" class="validation-advice">
+        <?php echo __("Please click the button below<br />and complete the information."); ?>
+    </div>
+
+    <button id="pagarme-modal-fill-info-button" class="button" type="button">
+        <span>
+            <span>
+                <?php echo $this->getButtonText(); ?>
```
Comment (devdrops) [inline app/design/frontend/base/default/template/pagarme/form/pagarme_modal.phtml]: Remember to always use the `__` translation resource :wink:

File: app/design/frontend/base/default/template/pagarme/form/pagarme_modal.phtml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            </span>
+        </span>
+    </button>
+</ul>
+
+<script type="text/javascript" id="pagarme-modal">
+    var checkoutButton = $('pagarme-modal-fill-info-button')
+
+    if ($('pagarme_modal_payment_method').value !== "") {
+      checkoutButton.remove();
```
Comment (devdrops) [inline app/design/frontend/base/default/template/pagarme/form/pagarme_modal.phtml]: It's good practices keeping a indentation pattern (2 or 4 spaces) :wink:

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php [hunk 4]
```diff
@@ -99,30 +102,57 @@ public function setCustomer(Mage_Customer_Model_Customer $customer)
         $this->customer = $customer;
     }
 
+    private function getPostbackUrl()
+    {
+        $activePaymentMethods = array_map('trim', explode(',', $this->getAvailablePaymentMethods()));
+        $isCreditCardActive = in_array('credit_card', $activePaymentMethods);
+        $isBoletoActive = in_array('boleto', $activePaymentMethods);
+
+        if ($isCreditCardActive && $isBoletoActive) {
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Form/Modal.php]: `else` is evil! Maybe we can change this structure, as below:

```php
if (($isCreditCardActive && $isBoletoActive)
    || (!$isCreditCardActive && !$isBoletoActive)
) {
    return '';
}

if ($isCreditCardActive) {
    return $this->getCreditCardPostbackUrl();
}

return $this->getBoletoPostbackUrl();
```

File: app/code/community/PagarMe/Modal/Block/Sales/RateAmount.php [hunk 5]
```diff
@@ -25,15 +25,14 @@ public function initTotals()
...
-                'field' => 'pagarme_checkout_rate_amount',
+                'code' => 'pagarme_modal_rate_amount',
+                'field' => 'pagarme_modal_rate_amount',
                 'value' => $rateAmount,
                 'label' => 'Interest Fee',
             ]);
 
             $this->getParentBlock()->addTotalBefore($total, 'grand_total');
         }
-
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Sales/RateAmount.php]: It's best practices keeping a extra line before a `return` statement.

**Comment by leonampd** [comment]
@devdrops Thank's for the comments. We'll fix the problems and send to you again soon. xD

File: app/code/community/PagarMe/Core/etc/config.xml (hunk 1 above)
Comment (leonampd) [reply app/code/community/PagarMe/Core/etc/config.xml]: Hey @devdrops are you talking about `checkout_capture_customer_data`? If yes, look at some lines above. This node was duplicated. Also, we changed to `modal_capture_custom_data` 😉

Comment (devdrops) [reply app/code/community/PagarMe/Core/etc/config.xml]: Nope! I noticed that the `checkout_button_text` option was removed.

Comment (leonampd) [reply app/code/community/PagarMe/Core/etc/config.xml]: Ah, ok! No this option still available but, we changed the name to `modal_button_text` some lines above.

File: app/code/community/PagarMe/CreditCard/Block/Form/Creditcard.php [hunk 6]
```diff
@@ -13,7 +13,7 @@ protected function _construct()
 
     public function getEncryptionKey()
     {
-        return Mage::getStoreConfig('payment/pagarme_settings/general_encryption_key');
+        return Mage::getStoreConfig('payment/pagarme_configurations/general_encryption_key');
```
Comment (devdrops) [inline app/code/community/PagarMe/CreditCard/Block/Form/Creditcard.php]: 86 columns :cry:

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php [hunk 7]
```diff
@@ -32,7 +32,7 @@ protected function _construct()
      */
     public function getEncryptionKey()
     {
-        return Mage::getStoreConfig('payment/pagarme_settings/general_encryption_key');
+        return Mage::getStoreConfig('payment/pagarme_configurations/general_encryption_key');
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Form/Modal.php]: 86 columns :cry:

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php [hunk 8]
```diff
@@ -99,30 +102,64 @@ public function setCustomer(Mage_Customer_Model_Customer $customer)
...
+    {
+        $activePaymentMethods = array_map('trim', explode(',', $this->getAvailablePaymentMethods()));
+        $isCreditCardActive = in_array('credit_card', $activePaymentMethods);
+        $isBoletoActive = in_array('boleto', $activePaymentMethods);
+
+        $postbackUrl = '';
+        if (($isCreditCardActive && $isBoletoActive)
+            || (!$isCreditCardActive && !$isBoletoActive)
+        ) {
+            $postbackUrl = '';
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Form/Modal.php]: If `$postbackUrl` is already defined as `''`, there is no reason to do it again :wink:

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php [hunk 9]
```diff
@@ -99,30 +102,64 @@ public function setCustomer(Mage_Customer_Model_Customer $customer)
...
+        return Mage::getBaseUrl() . 'pagarme_core/transaction_boleto/postback';
+    }
+
     /**
      * @return string
      */
     public function getAvailablePaymentMethods()
     {
-        return Mage::getStoreConfig('payment/pagarme_settings/checkout_payment_methods');
+        return Mage::getStoreConfig('payment/pagarme_configurations/modal_payment_methods');
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Form/Modal.php]: 86 columns :cry:

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php [hunk 10]
```diff
@@ -99,30 +102,64 @@ public function setCustomer(Mage_Customer_Model_Customer $customer)
...
     }
 
     public function hasFixedDiscountOnBoleto()
     {
-        return Mage::getStoreConfig('payment/pagarme_settings/boleto_discount_mode')
+        return Mage::getStoreConfig('payment/pagarme_configurations/boleto_discount_mode')
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Form/Modal.php]: To `==` or not to `===`, that is my question!

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php [hunk 11]
```diff
@@ -99,30 +102,64 @@ public function setCustomer(Mage_Customer_Model_Customer $customer)
...
             == PagarMe_Core_Model_System_Config_Source_BoletoDiscountMode::FIXED_VALUE;
     }
 
     private function hasPercentageDiscountOnBoleto()
     {
-        return Mage::getStoreConfig('payment/pagarme_settings/boleto_discount_mode')
+        return Mage::getStoreConfig('payment/pagarme_configurations/boleto_discount_mode')
```
Comment (devdrops) [inline app/code/community/PagarMe/Modal/Block/Form/Modal.php]: To `==` or not to `===`, that is my question!

File: app/design/frontend/base/default/template/pagarme/form/pagarme_modal.phtml [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        <span>
+            <span>
+                <?php echo __($this->getButtonText()); ?>
+            </span>
+        </span>
+    </button>
+</ul>
+
+<script type="text/javascript" id="pagarme-modal">
+    var checkoutButton = $('pagarme-modal-fill-info-button')
```
Comment (devdrops) [inline app/design/frontend/base/default/template/pagarme/form/pagarme_modal.phtml]: Start with 4 and everything else with 2 spaces === :cry:

File: app/code/community/PagarMe/Core/etc/config.xml (hunk 1 above)
Comment (henriquekano) [reply app/code/community/PagarMe/Core/etc/config.xml]: Also, there were two "checkout_button_text" in the original

File: app/code/community/PagarMe/Modal/Block/Form/Modal.php (hunk 11 above)
Comment (henriquekano) [reply app/code/community/PagarMe/Modal/Block/Form/Modal.php]: @devdrops: not sure if that answers, but the line below does compares with a constant. Isn't it enough?

File: app/code/community/PagarMe/Modal/etc/config.xml [hunk 13]
```diff
@@ -1,66 +1,66 @@
 <?xml version="1.0"?>
 <config>
     <modules>
-        <PagarMe_Checkout>
+        <PagarMe_Modal>
             <version>2.0.0</version>
```
Comment (henriquekano) [inline app/code/community/PagarMe/Modal/etc/config.xml]: Shouldn't the version be bumped?

**Pull request merged**